lstchain/calib/camera/r0.py

```diff
@@ -123,99 +123,101 @@ def __init__(self, **kwargs):
 
     def calibrate(self, event):
         for tel_id in event.r0.tels_with_data:
-            self.subtract_pedestal(event)
-            self.time_lapse_corr(event)
-            self.interpolate_spikes(event)
+            self.subtract_pedestal(event, tel_id)
+            self.time_lapse_corr(event, tel_id)
+            self.interpolate_spikes(event, tel_id)
 
-            event.r1.tel[self.tel_id].trigger_type = event.r0.tel[self.tel_id].trigger_type
-            event.r1.tel[self.tel_id].trigger_time = event.r0.tel[self.tel_id].trigger_time
+            event.r1.tel[tel_id].trigger_type = event.r0.tel[tel_id].trigger_type
+            event.r1.tel[tel_id].trigger_time = event.r0.tel[tel_id].trigger_time
 
             samples = event.r1.tel[tel_id].waveform[:, :, self.r1_sample_start:self.r1_sample_end]
             event.r1.tel[tel_id].waveform = samples.astype('float32') - self.offset
 
-
-    def subtract_pedestal(self, event):
+    def subtract_pedestal(self, event, tel_id):
         """
         Subtract cell offset using pedestal file.
         Fill the R1 container.
         Parameters
         ----------
         event : `ctapipe` event-container
+        tel_id : id of the telescope
         """
-        n_modules = event.lst.tel[self.tel_id].svc.num_modules
+        n_modules = event.lst.tel[tel_id].svc.num_modules
 
         for nr_module in range(0, n_modules):
-            self.first_cap_array[nr_module, :, :] = self._get_first_capacitor(event, nr_module)
+            self.first_cap_array[nr_module, :, :] = self._get_first_capacitor(event, nr_module, tel_id)
 
-        expected_pixel_id = event.lst.tel[self.tel_id].svc.pixel_ids
-        samples = np.copy(event.r0.tel[self.tel_id].waveform)
+        expected_pixel_id = event.lst.tel[tel_id].svc.pixel_ids
+        samples = np.copy(event.r0.tel[tel_id].waveform)
         samples.astype('int16')
         samples = subtract_pedestal_jit(
             samples,
             expected_pixel_id,
             self.first_cap_array,
             self.pedestal_value_array,
             n_modules)
-        event.r1.tel[self.tel_id].trigger_type = event.r0.tel[self.tel_id].trigger_type
-        event.r1.tel[self.tel_id].trigger_time = event.r1.tel[self.tel_id].trigger_time
-        event.r1.tel[self.tel_id].waveform = samples[:, :, :]
+        event.r1.tel[tel_id].trigger_type = event.r0.tel[tel_id].trigger_type
+        event.r1.tel[tel_id].trigger_time = event.r0.tel[tel_id].trigger_time
+        event.r1.tel[tel_id].waveform = samples[:, :, :]
```

Comment:
Why `samples[:,:,:]` and not just `samples` ?
Nothing wrong there but the reading is not simplified

Reply:
Right, thanks Thomas